testutil/base.go

```diff
@@ -62,8 +62,19 @@ func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
 
+// Backup a single element before further mocking.
+func Backup[T any](mockable *T) (restore func()) {
```

Comment:
Wooooot. Thank you l! I had this in the back of my head every time I saw the old mock being used.